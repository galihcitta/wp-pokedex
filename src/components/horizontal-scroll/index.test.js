import React from "react";
import HorizontalScroll from "./index";

let element;
let elementShallow;
beforeEach(() => {
    element = (
        <HorizontalScroll>
            <div test-id={"data-one"}>data</div>
        </HorizontalScroll>
    );
    elementShallow = shallow(element);
});

describe("HorizontalScroll Component", () => {
    it("renders correct and create snapshot <HorizontalScroll/>", () => {
        expect(toJson(elementShallow)).toMatchSnapshot();
    });
    it("renders correct and check children component <HorizontalScroll/>", () => {
        expect(
            elementShallow.children().find({ "test-id": "data-one" }).length,
        ).toEqual(1);
    });
});
