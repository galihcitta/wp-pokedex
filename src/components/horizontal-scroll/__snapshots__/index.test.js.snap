// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalScroll Component renders correct and create snapshot <HorizontalScroll/> 1`] = `
<styled.div
  noPaddingTop={false}
  style={
    Object {
      "style": Object {},
    }
  }
>
  <div
    className="horizontal-scroll"
  >
    <div
      className="horizontal-scroll--list"
    >
      <div
        test-id="data-one"
      >
        data
      </div>
    </div>
  </div>
</styled.div>
`;
